engines/cryo/cryolib.cpp

```diff
@@ -0,0 +1,453 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "common/system.h"
+#include "common/events.h"
+#include "common/timer.h"
+
+#include "graphics/palette.h"
+
+#include "cryo/cryo.h"
+#include "cryo/cryolib.h"
+
+namespace Cryo {
+
+///// Mac APIs
+
+void SysBeep(int x) {
+}
+
+void FlushEvents(int16 arg1, int16 arg2) {
+}
+
+///// CLView
+
+View::View(int w, int h) {
+	void *buffer = (byte *)malloc(w * h);
+	if (buffer)
+		initDatas(w, h, buffer);
+	else
+		error("Unable to allocate view buffer");
+}
+
+View::~View() {
+	if (_bufferPtr)
+		free(_bufferPtr);
+}
+
+// Original name: CLView_SetSrcZoomValues
+void View::setSrcZoomValues(int x, int y) {
+	_zoom._srcLeft = x;
+	_zoom._srcTop = y;
+}
+
+// Original name: CLView_SetDisplayZoomValues
+void View::setDisplayZoomValues(int w, int h) {
+	_zoom._width = w;
+	_zoom._height = h;
+}
+
+// Original name: CLView_InitDatas
+void View::initDatas(int w, int h, void *buffer) {
+	_bufferPtr = (byte *)buffer;
+	_width = w;
+	_height = h;
+	_pitch = w;
+	_doubled = false;
+	_normal._srcLeft = 0;
+	_normal._srcTop = 0;
+	_normal._dstLeft = 0;
+	_normal._dstTop = 0;
+	_normal._width = w;
+	_normal._height = h;
+	_zoom._srcLeft = 0;
+	_zoom._srcTop = 0;
+	_zoom._dstLeft = 0;
+	_zoom._dstTop = 0;
+	_zoom._width = w;
+	_zoom._height = h;
+}
+
+// Original name: CLView_CenterIn
+void View::centerIn(View *parent) {
+	_normal._dstLeft = (parent->_width - _normal._width) / 2;
+	_normal._dstTop = (parent->_height - _normal._height) / 2;
+	_zoom._dstLeft = (parent->_width - _zoom._width) / 2;
+	_zoom._dstTop = (parent->_height - _zoom._height) / 2;
+}
+
+///// CLPalette
+uint16 gIntervalLast, gIntervalFirst, gIntervalSet;
+int16 gMacintize = 0;
+color_t black_palette[256];
+color_t last_palette[256];
+
```

Comment:
Seems like all these CLPalette_foo functions will be eventually moved into a CLPalette class... right?